pkgs/development/tools/rust/cargo-flash/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, rustPlatform, pkg-config, rustfmt, libusb1 }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-flash";
+  version = "0.6.0";
+
+  src = fetchFromGitHub {
+    owner = "probe-rs";
+    repo = pname;
+    rev = "v" + version; 
+    sha256 = "01zyxbykzbqfy6w0cmhhq10l1qqf4aqxvh4jngj1lqba5r9d7jsc";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
+  cargoSha256 = "0fpcglng00vfx5643y2dcjik29vl2rz9vv8dx67gnpbkdrhpfxcx";
+
+  nativeBuildInputs = [ pkg-config rustfmt ];
```

Comment:
Is `rustfmt` an actual build dependency, or is this just used for tests/maintainer scripts/etc.? If any of the latter are actually getting run, this should probably be in `checkInputs` instead; otherwise maybe it could be removed?

Reply:
```
error: failed to run custom build command for `probe-rs v0.6.2`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/probe-rs-d89c7bfce87f839f/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Failed to run rustfmt: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/cargo-flash-0.6.0-vendor.tar.gz/probe-rs/build.rs:23:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

builder for '/nix/store/0xxncag0sfxc8kqiiwc6m5zlslrrd9xm-cargo-flash-0.6.0.drv' failed with exit code 101
error: build of '/nix/store/0xxncag0sfxc8kqiiwc6m5zlslrrd9xm-cargo-flash-0.6.0.drv' failed
```

If I interprete https://github.com/probe-rs/probe-rs/blob/master/probe-rs/build.rs#L21 correctly, this indeed is an actual build dependency.